### ISSUE TITLE
array-bracket-spacing "always" not enforcing "singleValue" : true

### ISSUE THREAD

**Issue by bennymack** [eslint/eslint#8354]
ESLint 3.18.0
node v4.4.5

**What did you do? Please include the actual source code causing the issue.**
Fresh git clone. Update the first three tests in "tests/lib/rules/array-bracket-spacing.js" by removing all spaces. 

```js
        { code: "var foo = obj[1]", options: ["always"] },
        { code: "var foo = obj['foo'];", options: ["always"] },
        { code: "var foo = obj[[1,1]];", options: ["always"] },
```

```
eslint]$ npm test

> eslint@3.18.0 test /home/bbixby/repos/eslint
> node Makefile.js test

Validating Makefile.js
Validating JSON Files
Validating Markdown Files
Validating JavaScript files
Validating JavaScript test files
Validating rules

  [▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬]

  14904 passing (3m)
  1 failing

  1) array-bracket-spacing valid var foo = obj[[1,1]];:

      AssertionError: Should have no errors but had 2: [ { ruleId: 'array-bracket-spacing',
    severity: 1,
    message: 'A space is required after \'[\'.',
    line: 1,
    column: 15,
    nodeType: 'ArrayExpression',
    source: 'var foo = obj[[1,1]];',
    fix: { range: [Object], text: ' ' } },
  { ruleId: 'array-bracket-spacing',
    severity: 1,
    message: 'A space is required before \']\'.',
    line: 1,
    column: 19,
    nodeType: 'ArrayExpression',
    source: 'var foo = obj[[1,1]];',
    fix: { range: [Object], text: ' ' } } ]
      + expected - actual

      -2
      +0

      at testValidTemplate (lib/testers/rule-tester.js:9:11640)
      at Context.<anonymous> (lib/testers/rule-tester.js:9:17803)



=============================================================================
Writing coverage object [/home/bbixby/repos/eslint/coverage/coverage.json]
Writing coverage reports at [/home/bbixby/repos/eslint/coverage]
=============================================================================

=============================== Coverage summary ===============================
Statements   : 99.45% ( 12174/12241 ), 149 ignored
Branches     : 98.04% ( 9385/9573 ), 79 ignored
Functions    : 99.86% ( 2110/2113 ), 23 ignored
Lines        : 99.45% ( 12083/12150 )
================================================================================

START:
29 03 2017 09:15:40.451:INFO [karma]: Karma v1.5.0 server started at http://0.0.0.0:9876/
29 03 2017 09:15:40.456:INFO [launcher]: Launching browser PhantomJS with unlimited concurrency
29 03 2017 09:15:40.476:INFO [launcher]: Starting browser PhantomJS
29 03 2017 09:15:41.034:INFO [PhantomJS 2.1.1 (Linux 0.0.0)]: Connected on socket -_bpOWPwLjHVX4FyAAAA with id 7233825

Finished in 1.111 secs / 0.852 secs @ 09:15:42 GMT-0400 (EDT)
SUMMARY:
✔ 236 tests completed
npm ERR! Test failed.  See above for more details.
```

**What did you expect to happen?**

Expected all three to fail

**What actually happened? Please include the actual, raw output from ESLint.**

Only third test failed.

**Comment by vitorbal** [comment]
Thanks for the report. However, those tests don't fail because they don't contain any array brackets. You might be confusing the brackets in the member expression `foo['bar']` with the brackets in e.g., an array literal: `foo = [ 'bar' ]`.
I think the rule you are looking for is the [computed-property-spacing](http://eslint.org/docs/rules/computed-property-spacing) rule.

**Comment by bennymack** [comment]
Ahh! Great! Works for me. Sorry about that.
